purb/skip.go

```diff
@@ -0,0 +1,208 @@
+package purb
+
+import (
+	"fmt"
+	"github.com/dedis/crypto/random"
+)
+
+// Pick a uint32 uniformly at random
+func randUint32() uint32 {
+	return random.Uint32(random.Stream)
+}
+
+// Pick a random height for a new skip-list node from a suitable distribution.
+func skipHeight() int {
+	height := 1
+	for v := randUint32() | (1 << 31); v&1 == 0; v >>= 1 {
+		height++
+	}
+	return height
+}
+
+type skipNode struct {
```

Comment:
If I understood correctly, this is taken from @bford 's code somewhere's else in the library ? If so, maybe we could come up with something that does not require duplicate the code ?